Automattic/woocommerce-payments
# Fix not allowed page when clicking Set up WooPayments from Core and account is already onboarded

Fixes #7625

#### Description

Merchants currently observe an issue with Set up WooPayments task: when account is fully onboarded, the link redirect them to `payments/onboarding` which results in Not allowed page error
![image](https://github.com/Automattic/woocommerce-payments/assets/79862886/99ed7ebd-0600-46f9-a2bb-49eed25c1023)

#### Changes proposed in this Pull Request

I decided that the best experience for merchants will be to get redirected to `payments/overview` page if the account is fully onboarded. I'll create a tech debt issue for that if that sounds good!

**Important**
Tests are not added since there is no tests for `maybe_handle_onboarding` function. Creating tests for it seems like out of scope for this PR.

#### Testing instructions

* Onboard an account as builder (`develop` branch)
* Get back to WC Home and click Set up WooPayments
* Observe the error described in the PR
* Checkout the branch `fix/7625-fix-setup-payments-task-for-completed-accounts` 
* Click Set up WooPayments
* Get redirected to `payments/overview`

Bonus: test that Set up WooPayments link works well when we don't have onboarded account.

-------------------


**Post merge**



- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-6.8.0#fix-not-allowed-page-when-clicking-set-up-woopayments-from-core-and-account-is-already-onboarded_
